modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/BaseResolverFactoryBean.java

```diff
@@ -75,16 +74,4 @@ public void initialise() throws InitialisationException
     {
         LifecycleUtils.initialiseIfNeeded(valueResolver, muleContext);
     }
-
-    /**
-     * Propagates the disposal event to the
-     * underlying {@link #valueResolver} if it implements
-     * the {@link Disposable} interface
-     */
-    @Override
-    public void dispose()
```

Comment:
This was a hack to make sure that the value was disposed. Now that the lifecycle was fixed, the value will get the dispose() by regular means